sdk/cosmosdb/cosmos/test/internal/session.spec.ts

```diff
@@ -70,6 +74,95 @@ describe("New session token", function () {
   });
 });
 
+describe("Integrated Cache Staleness", async function (this: Suite) {
+  beforeEach(async function () {
+    await removeAllDatabases();
+  });
+  const dbId = addEntropy("maxIntegratedCacheTestDB");
+  const containerId = addEntropy("maxIntegratedCacheTestContainer");
+  const dedicatedGatewayMaxAge = 600000;
+  const client = new CosmosClient({
+    endpoint,
+    key: masterKey,
+    consistencyLevel: ConsistencyLevel.Eventual,
+    plugins: [
+      {
+        on: "request",
+        plugin: async (context, next) => {
+          it("Should check if the max integrated cache staleness header is set and the value is correct.", async function () {
+            if (context.headers["x-ms-consistency-level"]) {
+              if (context.resourceType == ResourceType.item) {
+                if (context.headers["x-ms-dedicatedgateway-max-age"]) {
+                  assert.strictEqual(
+                    context.headers["x-ms-dedicatedgateway-max-age"].valueOf(),
+                    dedicatedGatewayMaxAge
+                  );
+                } else {
+                  assert(
+                    context.headers["x-ms-dedicatedgateway-max-age"],
+                    "x-ms-dedicatedgateway-max-age is not set."
+                  );
+                  assert.ifError(context.headers["x-ms-dedicatedgateway-max-age"]);
+                }
+              } else {
+                assert(
+                  context.headers["x-ms-dedicatedgateway-max-age"],
+                  "Attempt to use x-ms-dedicatedgateway-max-age on a non-item request."
+                );
+                assert.ifError(context.headers["x-ms-dedicatedgateway-max-age"]);
+              }
+            } else {
+              assert(
+                context.headers["x-ms-consistency-level"],
+                "x-ms-consistency-level is not set."
+              );
+              assert.ifError(context.headers["x-ms-consistency-level"]);
+            }
+          });
+          const response = await next(context);
+          return response;
+        },
+      },
+    ],
+  });
+  const itemRequestFeedOptions = {
+    maxIntegratedCacheStalenessInMs: dedicatedGatewayMaxAge,
+  };
+  const { database } = await client.databases.createIfNotExists({
+    id: dbId,
+  });
+  const { container } = await database.containers.createIfNotExists({
+    id: containerId,
+  });
+
+  // Should pass with maxIntegratedCacheStalenessInMs and consistency level set.
+  await container.items.create({ id: "1" });
+  await container.item("1").read(itemRequestFeedOptions);
+
+  // Should pass with maxIntegratedCacheStalenessInMs and consistency level set.
+  // read document.
+  await container.items.readAll(itemRequestFeedOptions).fetchAll();
```

Comment:
Does this API translate into a changefeed like request? In this case I believe we do not honor that request option setting.

Reply:
I don't think this test is needed, but it should fail. The parameter will be ignored by the service.

Reply:
Correct. If sent, this test will fail, and the service will ignore the pram.